### PR TITLE
Ignore the upcoming jobs of member's friends

### PR DESCRIPTION
**Pull request by wknds**
A better way than the committed solution would be by using distinct(), but it is not supported.
```
# assignmentdao.py
@staticmethod
def upcomming_assignments_for_member(member):
return juntagrico.models.Assignment.objects.filter(member=member).filter(job__time__gte=timezone.now()).distinct('job')
```

>> DISTINCT ON fields is not supported by this database backend

### CODE REVIEW

**Comment by motu55** [comment]
Distinct on only works with postgres, since we do not want to limit juntagrico to postgres allone we have to find another solution

**Comment by motu55** [comment]
https://docs.djangoproject.com/en/2.2/ref/models/querysets/#select-related

**Comment by motu55** [comment]
Try to use this not in the dao but where do the to set to list thing togethwr with distinct

**Comment by wknds** [comment]
> 
> 
> Try to use this not in the dao but where do the to set to list thing togethwr with distinct

What is the the purpose of select-related in combination with list(set(list)) ? To improve the performance?

**Comment by motu55** [comment]
actually not in combination with but instead of
something like
`AssignmentDao.upcomming_assignments_for_member(member).select_related('job').distinct().order_by('time')`
might do the trick. less code and look less hacky and probably even better performance

**Comment by wknds** [comment]
if i select_related('job'), then the corresponding template cannot access job.type.get_name anymore. i think it has something to do with https://github.com/juntagrico/juntagrico/blob/4ee4daa0a8bcdeac83af70c074b41c2c52d154ec/juntagrico/entity/jobs.py#L141

**Comment by motu55** [comment]
actually select_related is missleading since it does not do what i expected it to do, so the solution was to put the cart in front of the horse and now it is working, please have a look at it and let me know if you understand the solution

File: juntagrico/dao/jobdao.py [hunk 1]
```diff
@@ -61,3 +61,7 @@ def get_pinned_jobs():
     def get_promoted_jobs():
         return RecuringJob.objects.filter(type__name__in=Config.promoted_job_types(),
                                           time__gte=timezone.now()).order_by('time')[:Config.promomted_jobs_amount()]
+
+    @staticmethod
+    def upcomming_jobs_for_member(member):
+        return Job.objects.filter(time__gte=timezone.now(), assignment__member=member).distinct()
```
Comment (wknds) [inline juntagrico/dao/jobdao.py]: Why is it possible to access 'assignment' from Job? I don't see any relationship in the entity classes.

Comment (motu55) [reply juntagrico/dao/jobdao.py]: short answer magic, the long answer ist that django creates reversable field in the entity of a foreign key. More about that here
https://docs.djangoproject.com/en/2.2/topics/db/queries/#backwards-related-objects

Comment (wknds) [reply juntagrico/dao/jobdao.py]: Cool. It makes sense now!

**Pull request merged**